vitrina/datasets/views.py

```diff
@@ -845,9 +875,71 @@ def form_valid(self, form):
             rep.save()
             self.object.save()
 
+        messages.success(self.request, _("Išteklius sukurtas sėkmingai"))
+
         return HttpResponseRedirect(self.get_success_url())
 
 
+class ResourceSubclassCreateView(
+    LoginRequiredMixin,
+    PermissionRequiredMixin,
+    RevisionMixin,
+    CreateView,
+):
+    model = Dataset
+    template_name = "vitrina/datasets/resource_form.html"
+    context_object_name = "dataset"
+    form_class = ResourceSubclassForm
+
+    def has_permission(self):
+        next_url = self.request.GET.get("next")
+        if next_url:
+            match = resolve(next_url)
+            if match.url_name == "request-datasets":
+                if request_id := match.kwargs.get("pk"):
+                    request_obj = get_object_or_404(Request, pk=request_id)
+                    return has_perm(self.request.user, Action.ASSIGN, request_obj)
+        organization = get_object_or_404(Organization, id=self.kwargs.get("pk"))
+        return has_perm(self.request.user, Action.CREATE, Dataset, organization)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        organization = get_object_or_404(Organization, id=self.kwargs.get("pk"))
+        context["organization"] = organization
+        context["organization_id"] = organization.pk
+        context["current_title"] = _("Pridėti duomenų išteklį")
+        context["form_title"] = _("Ištekliaus poklasis")
+        context["form_description"] = _(
+            "Pasirinkite norimą sukurti duomenų ištekliaus poklasį"
+        )
+        context["use_custom_radio"] = True
+
+        context["parent_links"] = {
+            reverse("home"): _("Pradžia"),
+            reverse("organization-list"): _("Organizacijos"),
+            reverse("organization-detail", args=[organization.pk]): organization.title,
+            reverse("dataset-list"): _("Duomenų rinkiniai"),
+            "": _("Pridėti duomenų išteklį"),
+        }
+        context["current_step"] = 1
+        context["current_percentage"] = 5
+        return context
```

Comment:
You can use `context.update({})` and update all of these fields in one line (will span multiple lines, but will execute one operation). Agent template views have an example, look at those:-)